File: 0201-bitwise-and-of-numbers-range/0201-bitwise-and-of-numbers-range.cpp
class Solution {
public:
    int rangeBitwiseAnd(int l, int r) {
         int ans=0;
        for(int i=0;i<31;i++){
             long long ele=(1LL<<i);
             if((long long)r-l+1<=ele){
                 bool check=true;
                if(  (l&ele)==0 || (r&ele)==0  )check=false;
                 if(check)ans+=ele;
             }
        }
         return ans;
    }
};

File: 0201-bitwise-and-of-numbers-range/0201-bitwise-and-of-numbers-range.java
class Solution {
    public int rangeBitwiseAnd(int l, int r) {
        int count=0;
        boolean flag=false;
        for(int i=0;i<31;i++){
            long ele=(1<<i);
            if((long)(r-l+1)<=ele){
                flag=true;
                if((l&ele)==0 || (r&ele)==0) flag=false;
                if(flag) count+=ele;
            }
        }
        return count;
      
    }
}